### DISCUSSION TITLE
exportTrailingSlash vs trailingSlash

### DISCUSSION THREAD

**Discussion by oliviertassinari** [vercel/next.js#15598] [Ideas]
We are excited to see the new [`trailingSlash`](https://nextjs.org/docs/api-reference/next.config.js/trailing-slash) option landing in v9.5. So far we have been using [`exportTrailingSlash={true}`](https://nextjs.org/docs/api-reference/next.config.js/exportPathMap#adding-a-trailing-slash) when deploying the website in production to a CDN. However, in development, we have `trailingSlash={false}`, which can lead to some issues for us, for instance, when a developer works on a piece of content for the documentation, he might copy & paste a local URL to link a different page of the documentation, if he takes it from his env, he will miss the leading dash, which leads to a 301/308 in production, not great for SEO.

My question comes down to: is there any link between `exportTrailingSlash` and `trailingSlash`? It seems that they should be used in conjunction. Maybe the documentation should make a link between the two?

The question started in https://github.com/mui-org/material-ui/pull/21975#issuecomment-665234068.

**Comment by timneutkens** [comment]
if you want to force trailing slashes the recommendation is to just use `trailingSlash: true` both in dev/prod as both environment can route correctly since 9.5. Previously exportTrailingSlash would break development.

**Reply by oliviertassinari** [reply]
Ok thanks, we still have to set `exportTrailingSlash: true, trailingSlash: true` to get `/autocomplete/index.html` generated, over `/autocomplete.html`.

**Reply by dpkpaa** [reply]
if i want to force url to be traillingSlash  for few pages not for all page what to do in that case
 suppose country page urls

**Comment by jwugit** [comment]
@timneutkens 
hi, not sure if this is the right place to ask about slashes but i think its related.

We have a edge case due to some SEO requirements from management level and I'm trying to figure out if there is a solution for this.

The requirements is to have links on the web pages with no slash.
So when using next/link, running on server or on local dev, this is fine and good.
But when running SSG export mode, all the links are generated with an ending slash.

The exportTrailingSlash or now trailingSlash allow next/link to not generate link with ending slash during export. The file names are `/somepage.html` when export.

Because of static hosting, we want the file names to be `/somepage/index.html`
When setting trailingSlash to true, we get the proper export file names, but using this option now makes the next/link generated links has an ending slash.

So is there currently any support to allow exporting in `/somepage/index.html` but at the same time allow for SSG next/link to not include the ending slash automatically due to trailingSlash = true?

Thanks,

**Reply by sathyanarayanan-v** [reply]
I was able to make this work but with a little extra steps. I created a build and static export with `trailingSlash:true` and saved `out` to a folder, let's say `static-export`. I again created a build and static export with `trailingSlash:false` and copied the `out` to same folder `static-export`. Thus my `static-export` folder had `/somepage/index.html` and a file `/somepage.html`. Hope this helps.
